DataFormats/SoATemplate/interface/SoALayout.h

```diff
@@ -410,14 +409,35 @@
       CPP_TYPE * BOOST_PP_CAT(NAME, _) EDM_REFLEX_SIZE(elements_) = nullptr;                                           \
       ,                                                                                                                \
       /* Eigen column */                                                                                               \
+      byte_size_type BOOST_PP_CAT(NAME, Stride_) = 0;                                                                  \
       size_type BOOST_PP_CAT(NAME, ElementsWithPadding_) = 0; /* For ROOT serialization */                             \
       CPP_TYPE::Scalar * BOOST_PP_CAT(NAME, _) EDM_REFLEX_SIZE(BOOST_PP_CAT(NAME, ElementsWithPadding_)) = nullptr;    \
-      byte_size_type BOOST_PP_CAT(NAME, Stride_) = 0;                                                                  \
   )
 // clang-format on
 
 #define _DECLARE_SOA_DATA_MEMBER(R, DATA, TYPE_NAME) BOOST_PP_EXPAND(_DECLARE_SOA_DATA_MEMBER_IMPL TYPE_NAME)
 
+#define _COPY_VIEW_COLUMNS_IMPL(VALUE_TYPE, CPP_TYPE, NAME)                                          \
+  _SWITCH_ON_TYPE(                                                                                   \
+      VALUE_TYPE, /* Scalar */                                                                       \
+      memcpy(BOOST_PP_CAT(this->metadata().addressOf_, NAME)(),                                      \
+             BOOST_PP_CAT(view.metadata().addressOf_, NAME)(),                                       \
+             sizeof(CPP_TYPE));                                                                      \
+      , /* Column */                                                                                 \
+      memcpy(BOOST_PP_CAT(this->metadata().addressOf_, NAME)(),                                      \
+             BOOST_PP_CAT(view.metadata().addressOf_, NAME)(),                                       \
+             view.metadata().size() * sizeof(CPP_TYPE));                                             \
+      , /* Eigen column */                                                                           \
+      for (unsigned int i = 0; i < CPP_TYPE::RowsAtCompileTime * CPP_TYPE::ColsAtCompileTime; i++) { \
+        memcpy(BOOST_PP_CAT(this->metadata().addressOf_, NAME)() +                                   \
+                   i * cms::soa::alignSize(this->elements_ * sizeof(CPP_TYPE::Scalar), alignment) /  \
+                       sizeof(CPP_TYPE::Scalar),                                                     \
```

Comment:
I think this may be incorrect if `alignment` is not a multiple of `CPP_TYPE::Scalar`.
Which should never happen for elementary types like `float` or `double`, but _might_ happens for aggregates like `struct { float x; float y; float z; }`.
Whether it makes sense to have an Eigen matrix of those, it's a different matter...

Anyway, I see that the same formula is already used above:
```c++
      BOOST_PP_CAT(NAME, Stride_) = cms::soa::alignSize(elements_ * sizeof(CPP_TYPE::Scalar), alignment)               \
        / sizeof(CPP_TYPE::Scalar);                                                                                    \
```

Maybe this can be simplified reusing `BOOST_PP_CAT(NAME, Stride_)` ?